pyoptsparse/pyOpt_optimization.py

```diff
@@ -137,7 +142,17 @@ def checkConName(self, conName):
             return validName
 
     def addVarGroup(
-        self, name, nVars, type="c", value=0.0, lower=None, upper=None, scale=1.0, offset=0.0, choices=None, **kwargs
+        self,
+        name: str,
+        nVars: int,
+        varType: str = "c",
```

Comment:
We changed the name of this parameter here since `type` is a built-in function. This is a backwards-incompatible change, so what we should do is check if `type` is passed in through `kwargs`, and raise warning + set `varType` to `type`. This can be removed in a future release.

Reply:
Are you planning on adding the deprecation warning to this PR?

Reply:
This has been added.